Tests/SourceKitLSPTests/BuildSystemTests.swift

```diff
@@ -155,14 +153,17 @@ final class BuildSystemTests: XCTestCase {
     let newSettings = FileBuildSettings(compilerArguments: args + ["-DFOO"])
     buildSystem.buildSettingsByFile[doc] = newSettings
 
-    let expectation = XCTestExpectation(description: "refresh")
-    let refreshedDiags = try await testClient.nextDiagnosticsNotification()
-    XCTAssertEqual(refreshedDiags.diagnostics.count, 0)
-    XCTAssertEqual(text, documentManager.latestSnapshot(doc)!.text)
-
     await buildSystem.delegate?.fileBuildSettingsChanged([doc])
 
-    try await fulfillmentOfOrThrow([expectation])
+    var receivedCorrectDiagnostic = false
+    for _ in 0..<Int(defaultTimeout) {
+      let refreshedDiags = try await testClient.nextDiagnosticsNotification(timeout: 1)
+      if refreshedDiags.diagnostics.count == 0, text == documentManager.latestSnapshot(doc)!.text {
```

Comment:
Why is the document snapshot even tested in these tests 🤔?

Reply:
Probably to check that we get no diagnostics for the expected source file contents and not, e.g. no diagnostics because the file is empty. I probably wouldn’t have checked it if I wrote the test but 🤷🏽

Reply:
But that's not even what that is testing, unless I'm misunderstanding it. All it's checking is that our state in document manager matches what we expect, which isn't really anything to do with diagnostics.

Reply:
Happy to remove the check if you care.

Reply:
Unless you can actually see a reason for it, IMO it just confuses what the test is testing. So yeah, I'd be in favor of removing.

Reply:
Happy to leave it, but can you throw in a comment saying *why* we're doing this? Preferably with a link to the radar about fixing this in clangd, since it really does seem like a bug.

Reply:
I haven’t been able to find a good clangd reproducer yet outside of this test case, so this might actually be a race within the test itself. Since I want to re-write these tests with the new test workspace infrastructure anyway, I’ll leave it as-is for now and will add a comment to a clangd bug if the issue still persists with the new test infrastructure.